### ISSUE TITLE
[Bug Report] OPDS Android app Librera Reader 404 book not found and web .../opds error

### ISSUE THREAD

**Issue by zFerry98** [hkalexling/Mango#82]
On the android app you can navigate the library (you only see the names of the series and the volumes, no covers) you can select volumes and chapters but as soon as you open one of them comes out 404 book not found. If you write on your browser / opds this error comes out:
![image](https://user-images.githubusercontent.com/17975351/86002996-734f2b00-ba11-11ea-8b77-8885f1b1f4a3.png)

EDIT1 Android Screen Librera:
![image](https://user-images.githubusercontent.com/17975351/86119942-ba055980-bad3-11ea-8320-1a400468d35b.png)
![image](https://user-images.githubusercontent.com/17975351/86119986-cbe6fc80-bad3-11ea-89d9-10bd67829e78.png)
![image](https://user-images.githubusercontent.com/17975351/86120007-d3a6a100-bad3-11ea-9875-8279085f7681.png)
![image](https://user-images.githubusercontent.com/17975351/86120101-fdf85e80-bad3-11ea-9b5c-5349f2efb37f.png)
Android Screen PocketBook:
![image](https://user-images.githubusercontent.com/17975351/86120156-136d8880-bad4-11ea-8937-8bb38119fce6.png)
![image](https://user-images.githubusercontent.com/17975351/86120191-24b69500-bad4-11ea-8f80-29da772ff4b3.png)
Visualization as soon as you click on the server name (mango) in theory you should see the series library.

**Environment (please complete the following information):**
 - OS: Windows and Android 9.0 Lineage OS Galaxy S5
 - Browser chrome
 - Mango Version 0.7.1

**Docker
- The `docker-compose.yml` file you are using, or your `.env` file. Same of issue #79 

While we're here ... To update the docker with new releases, without losing the configuration and all the settings. Which method do you recommend?

**Comment by short-circuit** [comment]
I am having the same issue with KyBook3 and iOS. Thought in the beginning it might be a reverse proxy problem, but OPDS doesn't work on LAN either. No errors in logs either until now.

**Comment by zFerry98** [comment]
I have tried 3 different apps with OPDS support via network (SAMBA and http) but only Librera makes you scroll through the series. The other apps load endlessly after entering the password (e.g. PocketBook). The http version of the base site instead works wonderfully (the one without OPDS).

**Comment by hkalexling** [comment]
You got the XML error on the `/opds` page because one of the manga names contains the ampersand character (`&`) and Mango didn't escape it properly. This will be fixed in the next version. Thanks for reporting!

The 404 error is a separate issue and I couldn't reproduce it. Can you access the page `/opds/book/[id]` in your browser?

To update the image, simply use `docker-compose pull mango` to download the new image and start it with `docker-compose up`. The user configuration and library are stored on the host system as mapped volumes so they won't be affected.

**Comment by zFerry98** [comment]
![image](https://user-images.githubusercontent.com/17975351/86152009-d919e080-baff-11ea-8940-ea65f59db364.png)
Volume1.

------------------------------------

![image](https://user-images.githubusercontent.com/17975351/86152143-049ccb00-bb00-11ea-84b9-89023e42165b.png)
Series list:

**Comment by hkalexling** [comment]
@zFerry98 

1. When loading the page, there should be an error message printed in your console. Could you copy and paste it here?
2. Can you open the "Volume 01" folder in the non-opds site (i.e., `/book/[the_same_id]` without the `/opds/` prefix)?

**Comment by zFerry98** [comment]
> @zFerry98
> 
> 1. When loading the page, there should be an error message printed in your console. Could you copy and paste it here?
    - I have enabled debug output:
![image](https://user-images.githubusercontent.com/17975351/86222595-c1396f80-bb86-11ea-8b8c-bdf76dac0ac6.png)
    - This instead putting info on output:
![image](https://user-images.githubusercontent.com/17975351/86223477-e24e9000-bb87-11ea-937c-62be6e6fd613.png)

LOL You just don't recognize .zips? Strange because I also have CBZs.
![image](https://user-images.githubusercontent.com/17975351/86223739-3b1e2880-bb88-11ea-95de-f5cdf5bd3ebb.png)


----------------------------------------------------------------------------------------------------------------------------------------------

> 2. Can you open the "Volume 01" folder in the non-opds site (i.e., `/book/[the_same_id]` without the `/opds/` prefix)?
    - Only OPDS doesn't work. The normal library works all:
![image](https://user-images.githubusercontent.com/17975351/86222212-3a849280-bb86-11ea-8b09-5a70ef526261.png)
![image](https://user-images.githubusercontent.com/17975351/86222385-74ee2f80-bb86-11ea-868d-7fd0acfc3598.png)

**Comment by hkalexling** [comment]
Ah, I see the problem. It looks like Alpine (the base OS of the docker image) doesn't provide a MIME types registry, and so Mango failed to get the MIME types of the archive files. I will push a new release that fixes this soon.

**Comment by hkalexling** [comment]
Both issues are now fixed in [v0.7.2](https://github.com/hkalexling/Mango/releases/tag/v0.7.2). Thanks for reporting!

**Comment by zFerry98** [comment]
As I imagined, the best option is to use a machine with linux 64 (like ubuntu server or distro lite) than the docker. Which is a mess having a windows server. The problem is that I have no x64 pc but only x32 to use as mules. You say it works if you use linux bash support mode in windows? That function that adds a minimum of functions to run the terminal and the basic functions of a linux subsystem.

**Comment by hkalexling** [comment]
@zFerry98 No it's not a Windows issue. The new release should work in your current setup (Docker on Windows) and the two issues you reported should be fixed now.

**Comment by zFerry98** [comment]
No it's not a Windows issue. The new release should work in your current setup (Docker on Windows) and the two issues you reported should be fixed now.

No, I explained myself wrong. I meant that being a software for linux rather than for windows, it is better to run it directly on a real OS than on virtual machines. Having no more than 16gb of ram, since I put dockers I am always at 80 90% used. Running the server directly or having to dedicate Core and ram to virtualize an x64 pc with OS is certainly a part of the slowness problems. For now I have closed some services that I no longer use to recover resources. Anyway, thanks for the quick support and perfetto👍
